lib/record_store/record.rb

```diff
@@ -18,6 +18,12 @@ def initialize(record)
     end
 
     def self.build_from_yaml_definition(yaml_definition)
+      case yaml_definition.fetch(:type)
```

Comment:
as discussed... let's put the type in to a variable  that we then reference here, and below...